vit_vqgan/modeling_vit_vqgan.py

```diff
@@ -345,8 +334,15 @@ def setup(self):
         )
         self.transformer = Transformer(self.config, dtype=self.dtype)
 
+    @nn.compact
     def __call__(self, hidden_states, deterministic: bool = True):
-        hidden_states = hidden_states + self.position_embeddings(self.position_ids)
+        assert hidden_states.ndim == 3
+        position_embeddings = self.param(
+            "pos_embedding",
+            jax.nn.initializers.normal(self.config.initializer_range),
+            (1, 1, hidden_states.shape[1], hidden_states.shape[2]),
```

Comment:
The shape doesn't look right here, think it should be `(1, num_patches,  self.config.hidden_size)`. 
`(1, hidden_states.shape[1], hidden_states.shape[2])` shoul also work but seems less readable.

Reply:
Ok, I use `hidden_states` because I'm cheating and don't have to think about the right dimension but giving expected dimensions can help prevent silent bugs